catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -111,6 +114,21 @@ public String getName() {
     return name;
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * <p>This will load the resource.
+   */
+  @Override
+  public byte[] getByteArray() throws IOException {
+    loadResource();
+    if (byteArray == null && inputStream != null) {
+      byteArray = IOUtils.toByteArray(inputStream);
```

Comment:
❗️ This is also dangerous for very large resources as suddenly the memory requirement will increase and could cause out of memory.

Reply:
@paouelle I was mostly following the example of https://github.com/codice/ddf/blob/02b8e64ce99e6e89406e38b1b502bfa630497354/catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/BinaryContent.java#L67
I agree ByteBuffer would be safer but wouldn't that also require us to pull the resource into memory? The ticket for this originally called for the getInputStream() method to be able to return a fresh input stream each time but I couldn't find a good way to do that without pulling the resource into memory or leaving a stream open that an implementation would need to close. If you have any ideas about how to do that I would appreciate them 🙂

Reply:
I think we have 2 issues here. The first being that using a `byte[]` makes the stream data susceptible from being changed without the knowledge of where it originated. The second is the memory impact.

When I look at the ticket, it says nowhere that we should expose a byte array. It just says that attempting to retrieve multiple times the same input stream can cause problems. Couldn't we return a brand new input stream each time? Could we adopt an approach where we cache locally (assuming it is coming from a remote site) and return streams from that each time `getInputStream()` is called? Or maybe a more dynamic way of achieving this.